### DISCUSSION TITLE
Support for the phpactor LSP

### DISCUSSION THREAD

**Discussion by morganmsrn** [zed-industries/zed#10547] [Feature Requests]
Hi.

Is there any way to switch or have a picker to change the LSP used for PHP from Intelephense to [phpactor](https://phpactor.readthedocs.io/en/master/) ?

Main advantages to phpactor are that the project is open-source and has some built-in extensions not offered by Intelephense. I myself use the [phpstan](https://phpactor.readthedocs.io/en/master/integrations/phpstan.html) extension a lot on Laravel projects.

If there is a way to add this, as an extension or any other way, I'll gladly look into it myself if pointed in the good direction !

Thanks.

_edit: add links_

**Comment by frob** [comment]
I think you can add whatever LSP you want. At least it implies that in the docs. https://zed.dev/docs/adding-new-languages?highlight=lsp#lsp

**Reply by frob** [reply]
More info can be found in the issue with the merge. https://github.com/zed-industries/zed/issues/5652

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
